config/proto_conversions_test.go

```diff
@@ -562,7 +590,7 @@ func TestServiceConfigToProto(t *testing.T) {
 			test.That(t, err, test.ShouldBeNil)
 			test.That(t, out, test.ShouldNotBeNil)
 
-			test.That(t, out, test.ShouldResemble, &tc.Conf)
+			test.That(t, out.String(), test.ShouldResemble, tc.Conf.String())
```

Comment:
Changed this to compare their string representations because the AssociatedResourceConfigs field doesn't play well with `ShouldResemble`, and I think this is a bit cleaner/easier than checking the fields individually like `validateComponent` does